mesonbuild/dependencies/base.py

```diff
@@ -246,8 +246,8 @@ def __init__(self, version: str, incdirs: T.List['IncludeDirs'], compile_args: T
         self.include_directories = incdirs
         self.compile_args = compile_args
         self.link_args = link_args
-        self.libraries = libraries
-        self.whole_libraries = whole_libraries
+        self.libraries = list(libraries)
+        self.whole_libraries = list(whole_libraries)
```

Comment:
If this is already a T.List, why do you need to convert it into one again?

Reply:
It's not, i've changed it to a Sequence. List is invariant, so a `List[a| b | c | d| e | f]` is *not* fulfilled by `List[a]`. Sequence, however, is co-variant. So `List[a]` does fulfill `Sequence[ a | b | c]`. And this means that every place that wants to pass a list doesn't have to cast or define their list as a n `[a | b | c| d | e | f]`. In this case that's like 8 places in the gnome module.